corehq/apps/reports/standard/tableau.py

```diff
@@ -43,8 +45,13 @@ def page_url(self):
     def dispatch(self, request, *args, **kwargs):
         if self.visualization is None:
             raise Http404()
+        if not self._authenticate_request(request):
```

Comment:
I would be better to put this if statement first. Otherwise you can craw requests to see what exists and what not.